app/data/mysql/account_store.go

```diff
@@ -138,6 +138,16 @@ func (db *AccountStore) SetLastLogin(id int) (bool, error) {
 	return ok(result, err)
 }
 
+func (db *AccountStore) SetTOTPSecret(id int, secret []byte) (bool, error) {
+	result, err := db.Exec("UPDATE accounts SET totp_secret = ?, totp_enabled = true WHERE id = ?", secret, id)
```

Comment:
I had this in here because originally I wasn't caching the secret in redis, I was just writing it here. I would go Generate Secret -> Store In DB -> Verify ->Flip Flag To `true`. It can definitely be derived.